File: 백준/Silver/10828. 스택/스택.py
import sys
N = int(sys.stdin.readline())
# N = int(input())

stack = []

for _ in range(N):
    word = sys.stdin.readline().split()
    cmd = word[0]

    if cmd == "push":
        push_num = word[1]
        stack.append(push_num)
    elif cmd == "pop":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif cmd == "size":
        print(len(stack))
    elif cmd == "empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif cmd == "top":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[len(stack)-1])

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>

using namespace std;

int main()
{

    int N;
    cin >> N;

    stack<int> s1;
    string cmd;
    int cmd_num;
    for (int i = 0; i < N; i++)
    {
        cin >> cmd;

        if (cmd == "push")
        {
            cin >> cmd_num;
            s1.push(cmd_num);
        }
        else if (cmd == "pop")
        {
            if (s1.size() == 0)
            {
                cout << "-1" << endl;
            }
            else
            {
                cout << s1.top() << endl;
                s1.pop();
            }
        }
        else if (cmd == "size")
        {
            cout << s1.size() << endl;
        }
        else if (cmd == "empty")
        {
            if (s1.empty())
            {
                cout << "1" << endl;
            }
            else
            {
                cout << "0" << endl;
            }
        }
        else if (cmd == "top")
        {
            if (s1.size() == 0)
            {
                cout << "-1" << endl;
            }
            else
            {
                cout << s1.top() << endl;
            }
        }
    }

    // cout << cmd_num << endl;

    return 0;
}